[[ Bug 23094 ]] Fix android native layer rect relative to parent

montegoulding opened the pull request:
This patch ensures that when an android native container layer rect changes the
rect of child views is reset relative to the parent.

runrevmark commented on docs/notes/bugfix-23094.md:
```diff
@@ -0,0 +1 @@
+# Ensure android native layer rects are updated when their parent vide is resized
```
`vide` should be `view` I think?

livecodeian commented on engine/src/native-layer-android.cpp:
```diff
@@ -141,6 +141,7 @@ bool MCNativeLayerAndroid::doPaint(MCGContextRef p_context)
 
 void MCNativeLayerAndroid::doSetViewportGeometry(const MCRectangle& p_rect)
 {
+	doSetGeometry(m_object->getrect());
```
@montegoulding I think this object rect lookup is unneccessary as the currently set rect of a native layer is accessible as the protected MCNativeLayer::m_rect class member.

montegoulding committed:
[[ Bug 23094 ]] Fix android native layer rect relative to parent

This patch ensures that when an android native container layer rect changes the
rect of child views is reset relative to the parent.

runrevmark commented:
@livecode-vulcan review ok 335b38f

livecode-vulcan commented:
💙 review by @runrevmark ok 335b38f291a6d9e7dcb672afb71762aee0cf654a

livecode-vulcan commented:
😞 test failure 335b38f291a6d9e7dcb672afb71762aee0cf654a

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/965)
* try-community-armv7-android-ndk16r15: [failure](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/373)
* try-community-arm64-android-ndk16r15: [failure](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/372)
* try-community-x86-android-ndk16r15: [error](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/371)
* try-community-x86_64-android-ndk16r15: [failure](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/372)
* try-community-js-emscripten-sdk1.35: [failure](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2490)
* try-community-universal-ios-iphoneos14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.4/builds/19)
* try-community-universal-ios-iphoneos14.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.1/builds/68)
* try-community-universal-ios-iphonesimulator14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.4/builds/19)
* try-community-universal-ios-iphonesimulator14.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.1/builds/66)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2517)
* try-community-x86-linux-debian8: [error](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2506)
* try-community-x86_64-linux-debian8: [failure](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2504)
* try-community-x86-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2541)
* try-community-x86_64-win32: [error](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2477)

runrevmark commented:
@livecode-vulcan retry 335b38f

livecode-vulcan commented:
😎 test success 335b38f291a6d9e7dcb672afb71762aee0cf654a

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/966)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/374)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/373)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/372)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/373)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2491)
* try-community-universal-ios-iphoneos14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.4/builds/20)
* try-community-universal-ios-iphoneos14.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.1/builds/69)
* try-community-universal-ios-iphonesimulator14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.4/builds/20)
* try-community-universal-ios-iphonesimulator14.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.1/builds/67)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2518)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2507)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2505)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2542)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2478)
